Revert "libvirt: 4.5.0 -> 4.6.0" and attempted fixes

xeji opened the pull request:
###### Motivation for this change

Libvirt 4.6.0 is broken (runtime errors). The cause seems to be a migration from yajl to jansson for JSON support. Upstream has reverted this after 4.6.0. Instead of applying a large number of patches, 
let's revert to libvirt 4.5.0 (which worked fine) and wait for the next upstream release.

This reverts the version bump and some unsuccessful attempts to fix it.

See also discussion  in #45409 . 

###### Things done

Built on NixOS.

---

cc @volth @LnL7 @fpletz @pikajude @VShell

xeji committed:
Revert "libvirt: fix dlopen("libjansson.so.4")"

This reverts commit 929a53f2814982247f34fa296f6c1dd0435caf20.

Upstream has reverted their move from yajl to jansson after 4.6.0
because of major issues. No point in trying to fix it.
Let's revert to libvirt 4.5.0 and wait for the next upstream release.

xeji committed:
Revert "libvirt: jansson required for qemu suppport"

This reverts commit df05344d960895d8b076341ca0d8a0c7333f24b4.

Upstream has reverted their move from yajl to jansson after 4.6.0
because of major issues.
Let's revert to libvirt 4.5.0 and wait for the next upstream release.

xeji committed:
Revert "libvirt: 4.5.0 -> 4.6.0 (#44566)"

This reverts commit 88352408958d905c77ad0461036196282c39cf69.

4.6.0 is quite broken. Let's wait for the next release.

xeji commented:
@GrahamcOfBorg build libvirt

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=d83f919b-a057-4cb3-9945-4fc02a5659e7)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
Removing /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/share/gtk-doc
rmdir: failed to remove '/nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/share': Directory not empty
gzipping man pages under /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/share/man/
/nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip is /nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip
stripping (with command /nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip and flags -S) in /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/lib  /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/libexec  /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/bin  /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/sbin
patching script interpreter paths in /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0
/nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/zbhk0jhc9y8hmbcax9nvddbik3rgsary-bash-4.4-p23/bin/sh"
/nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/zbhk0jhc9y8hmbcax9nvddbik3rgsary-bash-4.4-p23/bin/sh"
moving /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/sbin/* to /nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0/bin
/nix/store/vv4mpc1kl4zl985r2g4n5y3jda3ca5cz-libvirt-4.5.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=b2335a43-4591-4672-a402-5075d66bb42c)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/var/lib/rc.d/init.d/libvirtd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/var/lib/rc.d/init.d/virtlockd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/var/lib/rc.d/init.d/libvirt-guests: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/var/lib/rc.d/init.d/virtlogd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0...
moving /nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/sbin/* to /nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0/bin
/nix/store/rad9d4ip10a4q1wc0akp2nnbc6irpfy6-libvirt-4.5.0
```
</p></details>

volth commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -60,15 +60,12 @@ in stdenv.mkDerivation rec {
     "--localstatedir=/var"
     "--sysconfdir=/var/lib"
     "--with-libpcap"
-    "--with-qemu"
```
I'd keep ```--with-qemu```, 
although it is default in ```4.5.0``` but conditional in ```4.6.0``` and possible ```4.7.0``` and it is better to be sure we build ```libvirt``` with ```qemu``` than to let ```pkg-config``` decide

volth replied:
I mean, on ```4.5.0 -> 4.7.0``` we likely will hit again this issue https://github.com/NixOS/nixpkgs/pull/44566#issuecomment-412222753

xeji replied:
Thanks, we may as well do that now :+1:

volth commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -16,30 +16,30 @@ let
   ] ++ optionals (!buildFromTarball) [
     libtool autoconf automake
   ] ++ optionals stdenv.isLinux [
-    libpciaccess lvm2 utillinux systemd libnl numad zfs
```
One ```lvm2``` is enough.

xeji committed:
libvirt: explicitly configure qemu support

it is enabled by default in 4.5.0 but conditional in 4.6.0 so
let's already put it in here to avoid issues on the next bump.

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=4c12e6a7-fcbb-418f-9a6f-693c4dd56f2f)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/var/lib/rc.d/init.d/libvirt-guests: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/var/lib/rc.d/init.d/virtlockd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/var/lib/rc.d/init.d/libvirtd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/var/lib/rc.d/init.d/virtlogd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0...
moving /nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/sbin/* to /nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0/bin
/nix/store/w76qmybq7w5iwx635702zf5wvpmvr1rg-libvirt-4.5.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=2df527f6-9832-4cc5-b28a-96eb8f8a46ee)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/var/lib/rc.d/init.d/libvirt-guests: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/var/lib/rc.d/init.d/virtlockd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/var/lib/rc.d/init.d/libvirtd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/var/lib/rc.d/init.d/virtlogd: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/6v88ick1cxnn5g91m8qrrqww0lrlr27x-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0...
moving /nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/sbin/* to /nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0/bin
/nix/store/315hmjfgi8ziqq7yriv66da0zqm3bb8z-libvirt-4.5.0
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=383324eb-f260-4b15-bf6c-f7009ffc6434)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
Removing /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/share/gtk-doc
rmdir: failed to remove '/nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/share': Directory not empty
gzipping man pages under /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/share/man/
/nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip is /nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip
stripping (with command /nix/store/5im29qx9rni3w8l3j8i2yr22sin55kxv-cctools-binutils-darwin/bin/strip and flags -S) in /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/lib  /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/libexec  /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/bin  /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/sbin
patching script interpreter paths in /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0
/nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/zbhk0jhc9y8hmbcax9nvddbik3rgsary-bash-4.4-p23/bin/sh"
/nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/zbhk0jhc9y8hmbcax9nvddbik3rgsary-bash-4.4-p23/bin/sh"
moving /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/sbin/* to /nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0/bin
/nix/store/7ywy4czarvqald1h99ipcqm9wj9n7z3s-libvirt-4.5.0
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45440&attempt_id=47551fa9-b4dd-4d8e-a13e-ec8eff082ca5)

Attempted: libvirt

<details><summary>Partial log (click to expand)</summary><p>

```
patching script interpreter paths in /nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/bin/virt-xml-validate: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/var/lib/rc.d/init.d/libvirt-guests: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/var/lib/rc.d/init.d/virtlockd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/var/lib/rc.d/init.d/libvirtd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/var/lib/rc.d/init.d/virtlogd: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/libexec/libvirt-guests.sh: interpreter directive changed from "/bin/sh" to "/nix/store/9hsgp158m12qi4dsvz6x6dxy5rkiind9-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0...
moving /nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/sbin/* to /nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0/bin
/nix/store/3nhafh72ai3irhbm4h7hq8wwdp64ji0r-libvirt-4.5.0
```
</p></details>

volth commented:
```4.7.0-rc2``` is out, so ```4.7.0``` will be released in 1-2 days
Will we upgrade, or keep ```4.5.0``` for ```nixpkgs-18.09``` ?

xeji commented:
Let's upgrade on master asap. We can always revert 18.09 to 4.5.0 in case of unexpected problems.
